src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1002,9 +1002,10 @@ def projection_complexities(self) -> List[Dict[PlanNodeCategory, int]]:
                         dependent_column_complexity = (
                             subquery_projection_name_complexity_map[dependent_column]
                         )
-                        projection_complexity[PlanNodeCategory.COLUMN] -= 1
                         projection_complexity = sum_node_complexities(
-                            projection_complexity, dependent_column_complexity
+                            projection_complexity,
```

Comment:
also update line 1000 to use proj.cumulative_node_complexity.copy() to be extra safe